test/as-needed-weak.sh

```diff
@@ -13,11 +13,11 @@ _start:
         call fn1@PLT
 EOF
 
-cat <<EOF | cc -o $t/b.so -shared -fPIC -Wl,-soname,libfoo.so -xc -
```

Comment:
I intentionally used the default linker instead of mold to build most shared object files to make sure that mold is interoperable with the system linker.

Reply:
@rui314 I think it would be nice to test both scenarios: link shared with mold & with ld and then test.